src/odemis/util/linalg.py

```diff
@@ -262,3 +262,7 @@ def generate_triangulation_points(max_dist: float, area_coords: Iterable[float])
         points = matrix.tolist()
 
     return points
+
+
+def check_overlap(range1: list, range2: list) -> bool:
```

Comment:
Please specify the type of the list: `List[float]`. 

Also, add a docstring explaining what this function computes.